### PR TITLE
Add Vivaldi Browser

### PR DESCRIPTION
**Pull request by 13r0ck**
I'm not certain if this is going to work. Particularly the arm64 packaging.

### CODE REVIEW

**Comment by jacobgkau** [comment]
This curiously doesn't seem to be on Flathub: https://flathub.org/apps/search/vivaldi 

I'm a fan of Vivaldi (it's my second or third favorite browser behind Brave), but is there any context behind adding this? Only asking since I thought we were trying to phase out of repo-proprietary (and as a web browser, I'd expect updates for this package to be both frequent and important.)

**Comment by 13r0ck** [comment]
Ya I checked flathub first too. Rather disapoimting honeslty. So this was my fallback. 

This is mostly because Vivaldi cares about their packaging on arm. So having Vivaldi available to users in pop shop. So this would be a good alternative without them having to add the PPA.

I didnt realize we were trying to phase out repo-proprietary. I guess I can just make another repo like we have `packaging-firefox` instead?

**Comment by jacobgkau** [comment]
I mean, I kind of disagree with that. We were trying to phase out of having to keep up-to-date with repackaging various apps, moving them into a bunch of different repos doesn't solve the problem.

**Comment by mmstick** [comment]
Getting discord and other packages packaged that way would also be good. Though we had some issues with packaging-vscode defaulting to a higher Pin-Priority than the VS Code PPA

**Comment by jacobgkau** [comment]
The only reason "packaging-x" became a thing was because repo-proprietary's build system was slow/inefficient and we got tired of waiting for all apps to be packaged just for a single app to be updated, right? Having all the third-party repackaging done in one place seems better than having it split across a bunch of different mostly-empty repos.

Plus, yeah, we incorrectly pin third-party apps above their own PPA default pins when we ship them as part of the actual Pop release repo.

File: suites/jammy.toml [hunk 1]
```diff
@@ -44,3 +44,17 @@ version = "1.19.1.2701-6327e27bf"
     [[direct.urls]]
     url = "https://downloads.plex.tv/plex-media-server-new/1.19.1.2701-6327e27bf/debian/plexmediaserver_1.19.1.2701-6327e27bf_amd64.deb"
     checksum = "aedee3bf69fc60fd86ca32555ce67ed6bc350d5552fd52df389d36c0b5875155"
+
+[[direct]]
+name = "vivaldi-stable"
+version = "5.6.2867.58-1"
+
+    [[direct.urls]]
+    url = "https://downloads.vivaldi.com/stable/vivaldi-stable_5.6.2867.58-1_amd64.deb"
```
Comment (jacobgkau) [inline suites/jammy.toml:53]: These URLs should have a ${version} placeholder so they don't need to be updated when the `version =` field above is updated.

**Review by jacobgkau — CHANGES_REQUESTED** [review]

**Comment by mmstick** [comment]
> The only reason "packaging-x" became a thing was because repo-proprietary's build system was slow/inefficient and we got tired of waiting for all apps to be packaged just for a single app to be updated, right?

Yeah, the gigabytes upon gigabytes of CUDA packages for each release really slowed down the generation of indexes since they weren't updated incrementally. Takes more effort to making a packaging-x repository but does benefit from using the same GitHub CI and repo-release workflow.

File: suites/jammy.toml (hunk 1 above)
Comment (13r0ck) [reply suites/jammy.toml]: Fixed, thanks for noticing that!

**Comment by 13r0ck** [comment]
Ya I do think we need to look into improving the build server to allow for repos that can build many distinct packages, without having to build all at once.

**Review by jacobgkau — APPROVED** [review]
The checksums are correct, so if we have an engineering consensus on supporting Vivaldi (not QA'ing the app itself since we have no control over it, but keeping the package up-to-date), then this looks correct. After merging, we would run the Jenkins apt-proprietary job, which creates the http://apt.pop-os.org/staging-proprietary/ repository. After testing from there to confirm the packages work (and the arm64 situation is as we expect), then we would run apt-proprietary-release to copy it to the actual proprietary repo.

Vivaldi's documentation suggests that installing the package will also install their repository: https://help.vivaldi.com/desktop/install-update/manual-setup-vivaldi-linux-repositories/ 

Their repo being installed is a reason we would want this to be in repo-proprietary and not repo-release (because updates will always hit their repo first, but the repo-release pin would keep our outdated version prioritized above their up-to-date version.) However, that also means we're not as likely to hear about it if the repo-proprietary package gets out-of-date, since users will see the updates they expect come in from Vivaldi's own repo instead. (I imagine an out-of-date package would mean a user installs the package, then immediately has another update available from Vivaldi's repo, which they may or may not choose to install right away.)

Their repo being installed would theoretically also mean that we could pull this from our repo later and not worry about users being orphaned on an outdated version, except for when pop-upgrade removes PPAs on the next Pop!_OS release upgrade (so pulling this later would probably still have to mean adding it to pop-transition, at least if a Flatpak becomes available later.)

**Comment by 13r0ck** [comment]
Oh ok cool! I think that it adding it's own repo is actually beneficial in this case. And yes, no QA should be required for Vivaldi, just that it is available for users to download from pop shop.

**Comment by 13r0ck** [comment]
@mmstick, I know this isn't perfect, but is it good enough for now?

**Comment by mmstick** [comment]
Yeah

**Comment by jacobgkau** [comment]
The package works to install and launch Vivaldi on amd64 using the staging-proprietary repo. I had to launch it twice before it started, not sure if that was a fluke. I did have an update available immediately after installing:

```
system76@pop-os:~$ apt policy vivaldi-stable
vivaldi-stable:
  Installed: 5.6.2867.58-1
  Candidate: 5.6.2867.62-1
  Version table:
     5.6.2867.62-1 500
        500 https://repo.vivaldi.com/stable/deb stable/main amd64 Packages
 *** 5.6.2867.58-1 500
        500 http://apt.pop-os.org/staging-proprietary jammy/main amd64 Packages
        500 https://repo.vivaldi.com/stable/deb stable/main amd64 Packages
        100 /var/lib/dpkg/status
```

Going to try and have someone check the arm64 packaging before releasing, since that was a concern.

**Comment by 13r0ck** [comment]
I just tested on the pi400, vivaldi works just fine. Very slow, but firefox is too

**Comment by n3m0-22** [comment]
I can confirm. It's working on my pi400 as well.

**Comment by 13r0ck** [comment]
Woot! @jacobgkau I will let you run the proprietary release if/when you are comfortable with this.

**Comment by jacobgkau** [comment]
Done!

**Pull request merged**